### PR TITLE
Support new hero image fields

### PR DESCRIPTION
**Pull request by davidrapson**
Reworks our API endpoints to support both new and re-designed hero images for all content types.

Required some reworking of–mostly older–api endpoints to use the same `getCommonFields` method.

### CODE REVIEW

File: lib/ContentHelpers.php [hunk 1]
```diff
@@ -18,11 +31,17 @@ public static function getCommonDetailFields(Entry $entry, $status, $locale)
...
             'dateUpdated' => $entry->dateUpdated,
             'availableLanguages' => EntryHelpers::getAvailableLanguages($entry->id, $locale),
+            // @TODO: Is url used anywhere?
+            'url' => $entry->url,
+            // @TODO: Some older pages use path instead of linkUrl in templates, update these uses and then remove this
+            'path' => $entry->uri,
             'linkUrl' => $entry->externalUrl ? $entry->externalUrl : EntryHelpers::uriForLocale($entry->uri, $locale),
             'title' => $entry->title,
-            'trailText' => $entry->trailText ?? null,
+            'trailText' => $entry->trailText ?? $entry->title,
```
Comment (mattandrews) [inline lib/ContentHelpers.php]: Is this right? Would it start showing double titles in places where we don't have trailtext?

Comment (davidrapson) [reply lib/ContentHelpers.php]: Not sure I follow, why would it show double titles?

Comment (mattandrews) [reply lib/ContentHelpers.php]: Instances like this:

image

That widget will render a card like this (if that element doesn't have trailtext):

image

I think it's better here to reflect the fact that no trailtext was provided (and let the template handle that, eg. some components might choose to use the title instead, but others won't).

Comment (davidrapson) [reply lib/ContentHelpers.php]: Interesting, that suggests we're using "trailText" to mean two different things. In some cases trail text means "alternate headline to use in place of the title on listings" and in others meaning "short line of description to use in *addition* to the title on listings". I'd rather standardise what we use this field for (but appreciate that might be for a later PR)

Comment (mattandrews) [reply lib/ContentHelpers.php]: 👍

Comment (davidrapson) [reply lib/ContentHelpers.php]: i.e. we got it right on the new updates section and should be using the `summary` here instead. Will roll this back for now though and make a ticket to review how we use this field more generally.

Comment (davidrapson) [reply lib/ContentHelpers.php]: Will roll this back here and raise a ticket in the backlog to sort this out more generally.

**Pull request merged**